### ISSUE TITLE
DragMarkers randomly disappear when dragging flutter_map

### ISSUE THREAD

**Issue by medinaStation24** [ibrierley/flutter_map_dragmarker#23]
Hello. I'm developing a Flutter web app and have encountered a serious bug when using flutter_map_dragmarker v4.0.2 and flutter_map 3.0.0. When I'm dragging the map sometimes and randomly dragmarkers will disappear from the map. When I drag back to where I was they will re-appear. Don't know why this is happening, the dragmarker that disappears isn't deleted, it is still there, it just disappears from the map. I tried using different icon for the drag marker but the issue persists. This occurs both on a desktop and on a mobile. Here are some screenshots depicting the bug:
![photo1](https://user-images.githubusercontent.com/111415905/195186459-35204c7f-b529-4004-ae6c-9f66945347e4.jpg)
![photo2](https://user-images.githubusercontent.com/111415905/195186478-1646f43b-9628-42c9-be1d-a317f8ae644c.jpg)

Using a Container with BoxDecoration and with text the number of the marker on a mobile phone:
![example2_photo1](https://user-images.githubusercontent.com/111415905/195186751-4491ded7-5acd-412b-8c40-742f279d694b.jpg)
![example2_photo2](https://user-images.githubusercontent.com/111415905/195186793-6e7b292a-56ce-4b71-bbc5-fe3eb6769dc7.jpg)

How FlutterMap and drag markers are used in my Scafold:
```
body: Stack(
	children: <Widget>[
	  FlutterMap(
		options: MapOptions(
			absorbPanEventsOnScrollables: false,
			center: LatLng(XX.XX, YY.YY),
			zoom: 8.5,
			maxZoom: 17.0,
			onTap: (tapPosition, location) => addMarker(location),
		),
		nonRotatedChildren: [
			AttributionWidget.defaultWidget(
				source: 'OpenStreetMap contributors',
				onSourceTapped: null,
			),
		],
		children: [
			TileLayer(
				urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
				userAgentPackageName: 'com.antoniskes.tsp_app',
			),
			DragMarkers(
			  markers: Data.locations,
			)
		],
	  ),
),
```

where locations are stored and the addMarker method:
```
  class Data{
	  static List<DragMarker> locations = List.empty(growable: true);
	  static int locationsCount = 0;
  }

void _addLocation(LatLng latLng){
	String markerText = Data.locationsCount.toString();
	TextStyle markerTextStyle = const TextStyle(fontSize: 26, fontWeight: FontWeight.bold, color: Colors.white);
	Color markerColor = markerText == '0' ? Colors.red : Colors.blue;

	Container markerIcon = Container(
	width: 60,
	height: 60,
	decoration: BoxDecoration(
	  borderRadius: BorderRadius.circular(32.0),
	  color: markerColor,
	),
	child: Center(
	  child: Text(
		markerText,
		style: markerTextStyle,
		textAlign: TextAlign.center,
	  ),
	),
	);

	DragMarker dragMarker = DragMarker(
	point: latLng,
	width: 60,
	height: 60,
	builder: (context) => markerIcon
	);
	Data.locations.add(dragMarker);
	Data.locationsCount++;

	setState(() {});
}
```

**Comment by ibrierley** [comment]
Thanks for the issue. Can you provide a minimal reproducable example that can be run, it's hard to have an idea just from those bits, although it gives the general idea.

**Comment by medinaStation24** [comment]
Yes, of course, thanks for looking into it. Here is an example main.dart code:

```
import 'package:flutter/material.dart';
import 'package:pointer_interceptor/pointer_interceptor.dart';
import 'package:flutter_map/flutter_map.dart';
import 'package:latlong2/latlong.dart';
import 'package:flutter_map_dragmarker/dragmarker.dart';
void main() {
  runApp(const MyApp());
}
class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'DragMarker',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'DragMarker disappears when dragging'),
    );
  }
}
class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);
  final String title;
  @override
  State<MyHomePage> createState() => _MyHomePageState();
}
class _MyHomePageState extends State<MyHomePage> {
  List<DragMarker> locations = List.empty(growable: true);
  int locationsCount = 0;
  void _addMarker(LatLng latLng){
      String markerText = locationsCount.toString();
      TextStyle markerTextStyle = const TextStyle(fontSize: 26, fontWeight: FontWeight.bold, color: Colors.white);
      // first location will have different color
      Color markerColor = markerText == '0' ? Colors.red : Colors.blue;
      Container markerIcon = Container(
        width: 60, height: 60,
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(32.0),
          color: markerColor,
        ),
        child: Center(
          child: Text(
            markerText,
            style: markerTextStyle,
            textAlign: TextAlign.center,
          ),
        ),
      );

      DragMarker dragMarker = DragMarker(
        point: latLng,
        width: 60, height: 60,
        builder: (context) => markerIcon
      );
      locations.add(dragMarker);
      locationsCount++;

      setState(() {});
  }
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
        centerTitle: true,
      ),
      body: Stack(
        children: <Widget>[
          FlutterMap(
            options: MapOptions(
                absorbPanEventsOnScrollables: false, // important to allow panning of markers
                center: LatLng(37.980352, 23.708522),
                zoom: 8.0,
                onTap: (tapPosition, location) => _addMarker(location),
            ),
            nonRotatedChildren: [
                AttributionWidget.defaultWidget(
                    source: 'OpenStreetMap contributors',
                    onSourceTapped: null,
                ),
            ],
            children: [
                TileLayer(
                    urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
                    userAgentPackageName: 'com.example.app',
                ),
                DragMarkers(markers: locations,),
            ],
          ),
          // LOCATIONS COUNTER
          PointerInterceptor(
            child: Text('Number of locations: $locationsCount',
              textAlign: TextAlign.center,
              overflow: TextOverflow.ellipsis,
              style: const TextStyle(
                backgroundColor: Colors.white,
                ),
            ),
          ),
        ],
      ),
    );
  }
}
```

**Comment by ibrierley** [comment]
Just a quick update, I think something odd is going wrong with the line

```if (!_boundsContainsMarker(mapState, marker)) continue;``` (prob line 31) in dragmarker.dart.

If you don't have a massive amount of markers, you could probably just comment out that line for a quick fix and include it alongside your other files, but I'll see if I can get to the bottom of why the bounds checking isn't working.

**Comment by ibrierley** [comment]
I've released 4.1.2, see if this fixes it for you.

I think the problem is that keys are needed when the list of markers is reordered now (as one will be removed if its out of bounds), so either a custom key can be included, or it will create one the same as the number in the list (so be wary if you have a list of markers that changes a lot, it may want to have a key included).

**Comment by medinaStation24** [comment]
> I've released 4.1.2, see if this fixes it for you.
> 
> I think the problem is that keys are needed when the list of markers is reordered now (as one will be removed if its out of bounds), so either a custom key can be included, or it will create one the same as the number in the list (so be wary if you have a list of markers that changes a lot, it may want to have a key included).

Thanks so much for the quick fix, this seems to fix the problem, for now I haven't encountered any issue.

**Comment by ibrierley** [comment]
Closing as hopefully fixed with Keys.
